### PR TITLE
Add `style` and `explode`

### PR DESCRIPTION
**Pull request by ackintosh**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Related issue: https://github.com/OpenAPITools/openapi-generator/issues/4039

In order to provide better collection serialization support, add `style` and `explode` attributes to CodegenParameter.

### CODE REVIEW

**Comment by ackintosh** [comment]
I've checked that the values are available in mustache files.

#### (PHP)

- api.mustache
```diff
        {{#allParams}}

+       {{#style}}var_dump('{{style}}');{{/style}}
+       var_dump({{shouldExplode}});

        {{/allParams}}
```

- FakeApi.php
```php
    protected function testQueryParameterCollectionFormatRequest($pipe, $ioutil, $http, $url, $context)
    {

        var_dump('form');
        var_dump(true);
```

**Comment by ackintosh** [comment]
cc @OpenAPITools/generator-core-team

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenParameter.java [hunk 1]
```diff
@@ -26,9 +26,9 @@
 public class CodegenParameter {
     public boolean isFormParam, isQueryParam, isPathParam, isHeaderParam,
             isCookieParam, isBodyParam, hasMore, isContainer,
-            secondaryParam, isCollectionFormatMulti, isPrimitiveType, isModel;
+            secondaryParam, isCollectionFormatMulti, isPrimitiveType, isModel, shouldExplode;
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenParameter.java]: @ackintosh thanks for the PR. What about using `isExplode` to follow the naming convention of other boolean flags?

cc @OpenAPITools/generator-core-team

Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenParameter.java]: Please update the following functions to include the new property:

https://github.com/OpenAPITools/openapi-generator/blob/df8d9d1188a117ec71af88790036c07d26179de6/modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenParameter.java#L100

https://github.com/OpenAPITools/openapi-generator/blob/df8d9d1188a117ec71af88790036c07d26179de6/modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenParameter.java#L181

https://github.com/OpenAPITools/openapi-generator/blob/df8d9d1188a117ec71af88790036c07d26179de6/modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenParameter.java#L254

https://github.com/OpenAPITools/openapi-generator/blob/df8d9d1188a117ec71af88790036c07d26179de6/modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenParameter.java#L323

Comment (ackintosh) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenParameter.java]: Updated the comment according to the renaming.
https://github.com/OpenAPITools/openapi-generator/pull/4042#issuecomment-541379088

**Comment by wing328** [comment]
CircleCI failure not related to this change.

**Pull request merged**